engine/gamesys/src/gamesys/resources/res_sound_data.cpp

```diff
@@ -41,6 +98,204 @@ namespace dmGameSystem
         return type;
     }
 
+    static dmResource::Result DestroyResource(SoundDataResource* resource)
+    {
+        // The sound data is reference counted
+        // The references are held here and also by any playing voices (e.g. like a long playing music track)
+        dmSound::SetSoundDataCallback(resource->m_SoundData, 0, 0);
```

Comment:
Since our sounds are fire and forget, we need to disconnect the resource and sound when this resource is deleted.